openmetadata-ui/src/main/resources/ui/cypress/common/SoftDeleteFlowUtils.js

```diff
@@ -0,0 +1,385 @@
+    );
+  }
+
+  interceptURL('GET', `/api/v1/tables/*/systemProfile*`, 'getSystemProfile');
+
+  cy.get('[data-testid="profiler"]').click();
+
+  verifyResponseStatusCode('@getSystemProfile', 200);
+
+  if (checkIsActionEnabled) {
+    cy.get('[data-testid="profiler-add-table-test-btn"]').should('be.visible');
+    cy.get('[data-testid="profiler-setting-btn"]').should('be.visible');
+  } else {
+    cy.get('[data-testid="table-profiler-container"]').then(($body) => {
+      const addTestButton = $body.find(
+        '[data-testid="profiler-add-table-test-btn"]'
+      );
+      const settingsButton = $body.find('[data-testid="profiler-setting-btn"]');
+
+      expect(addTestButton.length).to.equal(0);
+      expect(settingsButton.length).to.equal(0);
+    });
+  }
+};
+
+export const softDeletedEntityCommonChecks = ({
+  entityType,
+  firstTabKey,
+  entitySchemaName,
+  isLineageTabPresent,
+}) => {
+  const isTableEntity = entityType === DATA_ASSETS.tables;
+
+  // Check if all the edit actions are available for the entity
+  checkForEditActions({
+    entityType,
+    entitySchemaName,
+    verifyAPIResponse: true,
+  });
+
+  if (isLineageTabPresent) {
+    checkLineageTabActions({ verifyAPIResponse: true });
+  }
+
+  if (isTableEntity) {
+    checkForTableSpecificFields({ verifyAPIResponse: true });
+  }
+
+  // Soft delete the entity
+  cy.get('[data-testid="manage-button"]').click();
+  cy.get('[data-testid="delete-button-title"]').click();
+  cy.get('[data-testid="confirm-button"]').should('be.disabled');
+  cy.get('[data-testid="confirmation-text-input"]').type('DEL{enter}');
+  cy.get('#deleteTextInput_help').contains('Please type DELETE to confirm.');
+  cy.get('[data-testid="confirm-button"]').should('be.disabled');
+  cy.get('[data-testid="confirmation-text-input"]').clear().type('DELETE');
+  cy.get('[data-testid="confirm-button"]').click();
+  cy.get('.Toastify__close-button').click();
+  cy.get('[data-testid="deleted-badge"]').should('be.visible');
+
+  // Check if all the edit actions are disabled or removed for soft deleted entity
+  cy.get(`[data-testid="${firstTabKey}"]`).click();
+
+  checkForEditActions({
+    entityType,
+    entitySchemaName,
+    checkIsActionEnabled: false,
+  });
+
+  if (isLineageTabPresent) {
+    checkLineageTabActions({ checkIsActionEnabled: false });
+  }
+
+  if (isTableEntity) {
+    checkForTableSpecificFields({ checkIsActionEnabled: false });
+  }
+
+  cy.get('[data-testid="manage-button"]').click();
+
+  // only two menu options (restore and delete) should be present
+  cy.get('[data-testid="manage-dropdown-list-container"]').then(($body) => {
+    const editTier = $body.find(`[data-menu-id*="rc-menu"]`);
+
+    expect(editTier.length).to.equal(2);
+  });
+  cy.get(
+    '[data-testid="manage-dropdown-list-container"] [data-testid="restore-button"]'
+  );
+  cy.get(
+    '[data-testid="manage-dropdown-list-container"] [data-testid="delete-button"]'
+  );
```

Comment:
we are not performing any action here, we can remove this.

Reply:
Thanks for noticing this. Missed the assertions there. Will add those.